src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -741,7 +741,15 @@ public override bool Equals(object comparand)
 
         public override int GetHashCode()
         {
-            return (Name + "=" + Value + ";" + Path + "; " + Domain + "; " + Version).GetHashCode();
+            // Copied from Tuple.CombineHashCodes
+            int hash1 = Name.GetHashCode();
+            hash1 = ((hash1 << 5) + hash1) ^ Value.GetHashCode();
```

Comment:
@jamesqo In terms of app-compat issues regarding changing the implementation of GetHashCode(), it's ok to change it.

However, you should make sure that you have added tests to verify the principles of GetHashCode:

https://msdn.microsoft.com/en-us/library/system.object.gethashcode%28v=vs.110%29.aspx

>Two objects that are equal return hash codes that are equal. However, the reverse is not true: equal hash codes do not imply object equality, because different (unequal) objects can have identical hash codes.